Task 7

vkragel opened the pull request:
My Description here

TanyaGrek commented:
![image_2025-03-23_122053611](https://github.com/user-attachments/assets/79c8414d-4d82-4903-bb44-6a67eec5d42c)

vkragel commented:
@TanyaGrek Some error happens if you send null value for token. It's just a case that I didn't cover.
I don't think it's a reason to deduct -50 points)

TanyaGrek commented:
> @TanyaGrek Some error happens if you send null value for token. It's just a case that I didn't cover. I don't think it's a reason to deduct -50 points) I'll fix this for your peace of mind)

It's your frontend that sends null, not me. (Which I suppose should be covered by 401)
And you have CORS error as well.

so, with that being said, I can not verify that you have "correct" everything as they asked in the Evaluation criteria for task 7. I suppose -50 is a fair fee for not working product.

When I can test all cases successfully, I'll be happy to reevaluate your score.

vkragel commented:
@TanyaGrek I apologize for the misunderstanding. I see the issue with 500. I fixed it. But I don't quite understand what the problem is with CORS. I don't receive this error. Could you please recheck it again?

TanyaGrek commented:
> @TanyaGrek I apologize for the misunderstanding. I see the issue with 500. I fixed it. But I don't quite understand what the problem is with CORS. I don't receive this error. Could you please recheck it again?

Now all works, except the 401 error:

your FE always has a header Authorization
```
headers: { Authorization: `Basic ${token}` },
```
so even if the token is `null` the header will be `authorization: Basic null`

your BE returns **Unauthorized** only if there's no value for **authorizationToken**
``` 
 if (!authorizationToken) {
    logger.warn("Unauthorized: No authorization token provided");

    return generatePolicy("user", "Deny", event.methodArn);
  }
```

vkragel commented:
> > @TanyaGrek I apologize for the misunderstanding. I see the issue with 500. I fixed it. But I don't quite understand what the problem is with CORS. I don't receive this error. Could you please recheck it again?
> 
> Now all works, except the 401 error:
> 
> your FE always has a header Authorization
> 
> ```
> headers: { Authorization: `Basic ${token}` },
> ```
> 
> so even if the token is `null` the header will be `authorization: Basic null`
> 
> your BE returns **Unauthorized** only if there's no value for **authorizationToken**
> 
> ```
>  if (!authorizationToken) {
>     logger.warn("Unauthorized: No authorization token provided");
> 
>     return generatePolicy("user", "Deny", event.methodArn);
>   }
> ```

I got your point. I think, it's just better to fix it on FE side. Just send `Basic {token}` only if token exists. Otherwise, send "". I've fixed it. Please, check it again
